### PR TITLE
[IOTDB-2612] Fix cannot select compaction candidate after exception occurrence in previous compaction

### PR DESCRIPTION
**Pull request by THUMarkLau**
see [IOTDB-2612](https://issues.apache.org/jira/browse/IOTDB-2612)

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/cross/AbstractCrossSpaceCompactionTask.java [hunk 1]
```diff
@@ -95,7 +87,7 @@ public String toString() {
 
   @Override
   public void resetCompactionCandidateStatusForAllSourceFiles() {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/cross/AbstractCrossSpaceCompactionTask.java:89]:
```suggestion
  public void resetCompactionStatusForAllSourceFiles() {
```

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/AbstractInnerSpaceCompactionTask.java [hunk 2]
```diff
@@ -131,6 +127,6 @@ public String toString() {
 
   @Override
   public void resetCompactionCandidateStatusForAllSourceFiles() {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/AbstractInnerSpaceCompactionTask.java:129]:
```suggestion
  public void resetCompactionStatusForAllSourceFiles() {
```

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/cross/AbstractCrossSpaceCompactionTask.java (hunk 1 above)
Comment (THUMarkLau) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/cross/AbstractCrossSpaceCompactionTask.java:89]: resolve

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/AbstractInnerSpaceCompactionTask.java (hunk 2 above)
Comment (THUMarkLau) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/AbstractInnerSpaceCompactionTask.java:129]: resolve

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**